first-party/jib-native-image-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/nativeimage/JibNativeImageExtension.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle.extension.nativeimage;
+
+import com.google.cloud.tools.jib.api.JavaContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.FilePermissions;
+import com.google.cloud.tools.jib.gradle.ContainerParameters;
+import com.google.cloud.tools.jib.gradle.JibExtension;
+import com.google.cloud.tools.jib.gradle.extension.GradleData;
+import com.google.cloud.tools.jib.gradle.extension.JibGradlePluginExtension;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger.LogLevel;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+import com.google.common.base.Strings;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.gradle.api.Project;
+import org.gradle.internal.impldep.com.google.common.annotations.VisibleForTesting;
+
+public class JibNativeImageExtension implements JibGradlePluginExtension<Void> {
+
+  @VisibleForTesting
+  static Optional<String> getExecutableName(
```

Comment:
Ah open to seeing what is in the autconfig branch. It might be worth falling back to the main class the native gradle plugin specifies if the custom image name is not provided since it needs to run beforehand anyways.  Let me know if you've tried this out already.

Reply:
So the main issue I have is the unit test on that bit of code. The way the code works it seems you can't mock the object I use to get the info from the gradle plugin. I will fix the other changes then merge in the commit I have for this.

Reply:
@mpeddada1 So I have added this piece of code in. Manual tests show that this is working however there seems to be no way to mock this. Any thoughts on this point?

Reply:
Ok, so as a take away from this: I will pick the changes related to this out. And this is another dedicated PR.

Reply:
Actually, I just realized that we can have a MockNativeImageOptions class that implements `NativeImageOptions` and returns the desired `imageName` (here is an [example from our existing tests](https://github.com/GoogleContainerTools/jib-extensions/blob/7b3601e9b25625235423013102017e214c7e05cd/first-party/jib-quarkus-extension-gradle/src/test/java/com/google/cloud/tools/jib/gradle/extension/quarkus/MockComponentArtifactIdentifier.java#L22)) so we may not need to rely on integration tests.

But yes, let's add the `native-gradle-plugin` related changes in a follow-up PR.

Reply:
For visibility I did try something like that although there where some weird class no def errors in my case. But will have a further go.